app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationScreen.kt

```diff
@@ -123,6 +126,18 @@ fun ConversationScreen(
     val coroutineScope = rememberCoroutineScope()
     val showDialog = remember { mutableStateOf(ConversationScreenDialogType.NONE) }
 
+    val activity = LocalActivity.current
+    DisposableEffect(conversationInfoViewModel.conversationInfoViewState.screenshotCensoringEnabled) {
+        if (conversationInfoViewModel.conversationInfoViewState.screenshotCensoringEnabled) {
+            activity.window.addFlags(WindowManager.LayoutParams.FLAG_SECURE)
+        } else {
+            activity.window.clearFlags(WindowManager.LayoutParams.FLAG_SECURE)
+        }
+        onDispose {
+            activity.window.clearFlags(WindowManager.LayoutParams.FLAG_SECURE)
+        }
+    }
+
```

Comment:
I would expect this logic to be in the wire activity view model and apply to the whole app, not only the conversation screen

Reply:
I thought it was intended. That is, that only the conversation content is protected.

Is this really the case? If it isn't, then I'd join @MohamadJaara on thinking this should be moved to the `WireActivityViewModel`.